### PR TITLE
fix(key),test: simplify the input analysis code

### PR DESCRIPTION
**Pull request by knz**
This PR contains only the input simplification part of #511.

cc @muesli

### CODE REVIEW

File: key.go [hunk 1]
```diff
@@ -336,87 +337,77 @@ var keyNames = map[KeyType]string{
 }
 
 // Sequence mappings.
+//
+// Note: if you modify this map, do not forget to re-run `go generate`.
```
[the remark this answers is not in the document]
Comment (knz) [reply key.go]: Good point. Removed.

**Comment by knz** [comment]
@muesli this is ready again.

File: key_test.go [hunk 2]
```diff
@@ -191,34 +384,236 @@ func TestReadInput(t *testing.T) {
...
 			}
+		})
+	}
+}
 
-			for i, v := range msgs {
-				if m, ok := v.(KeyMsg); ok &&
-					m.String() != td.out[i].(KeyMsg).String() {
-					t.Fatalf(`expected a keymsg %q, got %q`, td.out[i].(KeyMsg), m)
+// randTest defines the test input and expected output for a sequence
```
Comment (muesli) [inline key_test.go:478]: Good enough for now, but I guess we could eventually replace the randomizer with a proper fuzzer?

Comment (knz) [reply key_test.go:478]: Yes that is reasonable. Do you want to do this right away? Is there something standard we can use?

Comment (muesli) [reply key_test.go:478]: Let's keep it as it is, the code looks and works fine for me. As soon as we bump Go to 1.18 we can start moving to the Go fuzzer that's part of the standard toolchain: https://go.dev/security/fuzz/

**Pull request merged**